Fix #7 for Python, CPP

sujanchegu opened the pull request:
Separate commits to fix #7 in Python, CPP. 

Checklist: 
My modifications generate no new warnings
I've commented my code wherever possible.

sujanchegu committed:
[CODE] Add check for invalid no. of start states

sujanchegu committed:
[CODE] Add check for invalid no. of start states for CPP

MadhavJivrajani commented on Python/dfa.py:
```diff
@@ -21,17 +21,30 @@ def __init__(self, alphabet):
+            if is_start:
+                self.__num_start_states += 1
+            
+            if self.__num_start_states != 1:
```
This condition won't work if the user decides to enter the start state after the addition of a first state. 

A better condition would be `self.__num_start_states > 1`

The same feedback goes for the c++ code as well.

MadhavJivrajani commented on C++/dfa.hpp:
```diff
@@ -17,6 +17,7 @@ class DFA {
   map<string, vector<tuple<string, string, string>>> transitions;
   map<string, State*> states;
   int rejected;
+  int __num_start_states;
```
__num_start_states should not be declared public, that's something that the class uses internally to check for the number of start states.

You put `__` in front of a class attribute/method in python to declare it as private. Afaik there's no convention as such in C++ since it has explicit `public` and `private` modes, so you can lose the double underscore in front of the name in the c++ code.

sujanchegu committed:
Modify changes requested to CPP

1. make num_start_states a private variable; 2. Change notequalto - greaterthan

sujanchegu committed:
Modify changes requested to Python

1. Change notequalto - greaterthan

MadhavJivrajani committed:
Update dfa.py
